src/coordination/coordinator_instance.cpp

```diff
@@ -214,78 +223,148 @@ auto CoordinatorInstance::ShowInstances() const -> std::vector<InstanceStatus> {
+        .cluster_role = get_coord_role(instance.coordinator_id, curr_leader),
+        .health = stringify_coord_health(instance),
+        .last_succ_resp_ms = raft_state_->CoordLastSuccRespMs(instance.coordinator_id).count(),
+
+    };
   };
 
-  auto instances_status = utils::fmap(raft_state_->GetCoordinatorInstances(), coord_instance_to_status);
+  return utils::fmap(raft_state_->GetCoordinatorInstances(), coord_instance_to_status);
+}
 
-  if (raft_state_->IsLeader()) {
-    auto const stringify_repl_role = [this](ReplicationInstanceConnector const &instance) -> std::string {
-      if (!instance.IsAlive()) return "unknown";
-      if (raft_state_->IsCurrentMain(instance.InstanceName())) return "main";
-      return "replica";
-    };
+auto CoordinatorInstance::GetAllInstancesStatusAsFollower() const -> std::vector<InstanceStatus> {
+  auto instances_status = GetCoordinatorsInstanceStatus();
+  spdlog::trace("Processed all coordinators, processing replication instances as follower.");
+  auto const stringify_inst_status = [raft_state_ptr =
+                                          raft_state_.get()](ReplicationInstanceState const &instance) -> std::string {
+    if (raft_state_ptr->IsCurrentMain(instance.config.instance_name)) {
+      return "main";
+    }
+    if (raft_state_ptr->HasMainState(instance.config.instance_name)) {
+      return "unknown";
+    }
+    return "replica";
+  };
 
-    auto const stringify_repl_health = [](ReplicationInstanceConnector const &instance) -> std::string {
-      return instance.IsAlive() ? "up" : "down";
-    };
+  auto process_repl_instance_as_follower =
+      [&stringify_inst_status](ReplicationInstanceState const &instance) -> InstanceStatus {
+    return {.instance_name = instance.config.instance_name,
+            .management_server = instance.config.ManagementSocketAddress(),  // show non-resolved IP
+            .bolt_server = instance.config.BoltSocketAddress(),              // show non-resolved IP
+            .cluster_role = stringify_inst_status(instance),
+            .health = "unknown"};
+  };
 
-    auto process_repl_instance_as_leader =
-        [&stringify_repl_role, &stringify_repl_health](ReplicationInstanceConnector const &instance) -> InstanceStatus {
-      return {.instance_name = instance.InstanceName(),
-              .management_server = instance.ManagementSocketAddress(),  // show non-resolved IP
-              .bolt_server = instance.BoltSocketAddress(),              // show non-resolved IP
-              .cluster_role = stringify_repl_role(instance),
-              .health = stringify_repl_health(instance),
-              .last_succ_resp_ms = instance.LastSuccRespMs().count()};
-    };
+  std::ranges::transform(raft_state_->GetReplicationInstances(), std::back_inserter(instances_status),
+                         process_repl_instance_as_follower);
+  spdlog::trace("Processed all replication instances as follower, returning report as follower.");
```

Comment:
This log, returning report to follower is actually a bit confusing. Do we really need it?

Reply:
It is easier to reason why something happened when customer asks I would say